components/eamxx/src/diagnostics/field_at_pressure_level.cpp

```diff
@@ -71,21 +82,36 @@ void FieldAtPressureLevel::compute_diagnostic_impl()
     auto fdat = view_Nd<mPack,2>("",f_data_src.extent_int(0),f_data_src.extent_int(1));
```

Comment:
Allocating memory on device at run time is not great. Can this helper view be created at init time? Perhaps when field_src is set? If not, you could have it as a member var, and do
```
if (m_fdat.size()==0) {
  m_dat = view_Nd<mPack,2>(...);
}
```
so that the allocation happens only once.